### DISCUSSION TITLE
Browse and replay cached songs

### DISCUSSION THREAD

**Discussion by nidyran** [z-huang/music#1216] [General]
I appreciate the caching feature in the application. However, I am wondering if I can list and play cached songs once I am offline. 

Currently, I check the history list to play cached songs, but this method is not always optimal as some songs may not be fully available in the cache. 

According to this thread `https://github.com/z-huang/InnerTune/discussions/993`, songs are downloaded in chunks. 

Is it possible to determine the list of songs where all the chunks are downloaded? 
Is there any planned feature regarding this topic? 

Finally, I would like to express my gratitude for developing this app!

**Comment by nvllz** [comment]
> but this method is not always optimal as some songs may not be fully available in the cache

You can manually download songs, which will cache all the chunks and store them in the app's data storage. I don't think there's a way to list fully cached songs, but there is for downloaded ones. Basically, downloading works like pre-caching, so you don't have to listen to a song to have it fully cached.
